search

holyjak opened the pull request:
Implement custom search; fixes #85

See #85, doc/adr/0019-use-custom-search.md, and the ns comment
in src/cljdoc/server/search/search.clj

## TODO in a follow-up PR

* Take download counts into account

martinklepsch commented on src/cljdoc/server/search/shared.clj:
```diff
@@ -0,0 +1,6 @@
+           (java.nio.file Paths)))
+
+(defn ^FSDirectory fsdir [index-dir]
+  (FSDirectory/open (Paths/get index-dir (into-array String nil))))
```
Unless there's a reason that this is in its own namespace I'd suggest to move it into the other namespace.

holyjak replied:
well, we need to use it in both in `search` and `artifact-indexer` ns. The only alternative is to duplicate it in both.

holyjak commented:
@martinklepsch I have addressed all the comments except "ensure perfect matches are scored higher" - I would prefer to do that one in a follow up as it may take a while.

Do you want to review the latest commit? I can then squash the two commits together, for a nicer history.

martinklepsch commented:
Oh and due to #233 being merged you'll need to rebase the branch. Sorry about that :)

holyjak commented on src/cljdoc/server/release_monitor.clj:
```diff
@@ -63,19 +65,24 @@
+                                       :group-id (:group_id to-build)
+                                       ;; NOTE: Ideally we would include also the old versions;
+                                       ;; but they will be re-added once the full re-import runs anyway
+                                       :versions [(:version to-build)]})))))
```
Not really since I don't truly understand what they do but I trust you and have changed it as suggested :)

martinklepsch replied:
`release-fetch-job-fn` retrieves new releases from Clojars and stores them into the database. It does this by looking up the last release added to the database and only adding the ones to the database that were added after that one. 

`build-queuer-job-fn` takes the oldest release from the same table that hasn't been built and initiates a build for it. This is how builds are automated. 

In short `release-fetch-job-fn` gets all releases and `build-queuer-job-fn` starts a build for a single release.

Hope this clears things up a bit :)
